src/main/charts/bitbucket/templates/shared-home-pv.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if .Values.volumes.sharedHome.persistentVolume.create }}
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ include "bitbucket.fullname" . }}-shared-home-pv
+  labels:
+  {{- include "bitbucket.labels" . | nindent 4 }}
+spec:
+  capacity:
+    storage: 1Gi
```

Comment:
this should be an entry in `values.yaml`

Reply:
ATM, it's not used and exists only to satisfy Kubernetes validation rules, adding it to values would be misleading.

Reply:
I agree with @kennymacleod about not including a PV. A couple of reasons here:
* like @kennymacleod said, flushing data when the chart is uninstalled isn't good
* it's common to setup up automatic PV provisioners in K8s clusters, so, we just create PVC and a provisioner will create a PV
* it's common that people who install and manage apps installed from Helm charts do not have permissions to create PVs which are requested from infra team on a per-case basis
* PV is this sort of an object that is not updated like stateful set or some configmap on helm upgrade. It's more like part of K8s infra to me even though it's an API object.It is also not related to the app itself.
* if we do want to include a PV template we need to make it all configurable then, not just limit to NFS. I mean let a user provide annotations (very important!) and PV spec in values.yaml so that a user can create a PV in AKS, EKS, GKE etc. If we do that, it'd be great not to track it in Helm after creation, i.e. not update/delete (maybe pre-install hook, or smth like that)

Reply:
Eugene, thank you for providing a real world perspective on the how charts should look like. I understand that with your experience with proper Kubernetes deployments, you're looking at it from a different angle than I do. I think we're in a slightly different scenario, since we're packaging an app that (at least for some cases) has a hard requirement on NFS, specifically version 3 with very specific mount options, otherwise it won't work correctly.

To address your points:
> like @kennymacleod said, flushing data when the chart is uninstalled isn't good
It's not flushed in case of NFS.

> it's common to setup up automatic PV provisioners in K8s clusters, so, we just create PVC and a provisioner will create a PV
That's fine, we support that now. It's not a reason to not make it easier to support an NFS server (potentially) not tied to Kubernetes.

> it's common that people who install and manage apps installed from Helm charts do not have permissions to create PVs which are requested from infra team on a per-case basis

I don't think it's a reason to not support PV creation. In the scenario you've outlined, the PV definition can be used by the infra team to create a PV with precisely the parameters that we require in BBS. The app team can then go the PVC route, which we also support.

> PV is this sort of an object that is not updated like stateful set or some configmap on helm upgrade. It's more like part of K8s infra to me even though it's an API object.It is also not related to the app itself.

I understand that this is true in general, but I don't think that's correct in this case, where we're more or less providing an representation of an external infra as a Kubernetes object specifically for the purposes of our deployment. For example, we need the PV to contain specific mount options for NFS (otherwise especially BBS will not function correctly). If we start supporting newer NFS versions or require different options, we need to update the PV definition with the chart. I think this makes perfect sense (caveat: for NFS only).

> we do want to include a PV template we need to make it all configurable then, not just limit to NFS.

For the time being, we support 'everything' through the PVCs. We also have a specific case that makes it easier to use NFS, which is the only supported option for BBS and simply has to be used there to have a supported setup. Personally, I think supporting PVCs and a special case for NFS PV is good enough until the feature requests start trickling in. And that's where the product managers will step in.